client/trino-client/src/main/java/io/trino/client/uri/TrinoUri.java

```diff
@@ -453,373 +153,457 @@ public URI getUri()
-        }
-        catch (RuntimeException e) {
-            throw new SQLException("Error setting up connection", e);
-        }
+    public Optional<Duration> getExternalAuthenticationTimeout()
+    {
+        return resolveOptional(EXTERNAL_AUTHENTICATION_TIMEOUT);
+    }
+
+    public Optional<List<ExternalRedirectStrategy>> getExternalRedirectStrategies()
+    {
+        return resolveOptional(EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS);
+    }
+
+    public KnownTokenCache getExternalAuthenticationTokenCache()
+    {
+        return resolveWithDefault(EXTERNAL_AUTHENTICATION_TOKEN_CACHE, KnownTokenCache.NONE);
+    }
+
+    public String getDnsResolverContext()
+    {
+        return resolveWithDefault(DNS_RESOLVER_CONTEXT, null);
+    }
+
+    public Optional<Class<? extends DnsResolver>> getDnsResolver()
+    {
+        return resolveOptional(DNS_RESOLVER);
+    }
+
+    public Optional<Boolean> getExplicitPrepare()
+    {
+        return resolveOptional(EXPLICIT_PREPARE);
+    }
+
+    public Optional<Boolean> getAssumeNullCatalogMeansCurrentCatalog()
+    {
+        return resolveOptional(ASSUME_NULL_CATALOG_MEANS_CURRENT_CATALOG);
     }
 
-    private static DnsResolver instantiateDnsResolver(Class<? extends DnsResolver> resolverClass, Optional<String> context)
+    public boolean isCompressionDisabled()
+    {
+        return resolveWithDefault(DISABLE_COMPRESSION, false);
+    }
+
+    public boolean isAssumeLiteralNamesInMetadataCallsForNonConformingClients()
+    {
+        return resolveWithDefault(ASSUME_LITERAL_NAMES_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS, false);
+    }
+
+    public boolean isAssumeLiteralUnderscoreInMetadataCallsForNonConformingClients()
+    {
+        return resolveWithDefault(ASSUME_LITERAL_UNDERSCORE_IN_METADATA_CALLS_FOR_NON_CONFORMING_CLIENTS, false);
+    }
+
+    public ZoneId getTimeZone()
+    {
+        return resolveWithDefault(TIMEZONE, ZoneId.systemDefault());
+    }
+
+    public Locale getLocale()
+    {
+        return resolveWithDefault(LOCALE, Locale.getDefault());
+    }
+
+    public Duration getTimeout()
+    {
+        return resolveWithDefault(TIMEOUT, Duration.valueOf("30s"));
+    }
+
+    public HttpLoggingInterceptor.Level getHttpLoggingLevel()
+    {
+        return resolveWithDefault(HTTP_LOGGING_LEVEL, HttpLoggingInterceptor.Level.NONE);
+    }
+
+    public Properties getProperties()
+    {
+        return properties;
+    }
+
+    public <V, T> T resolveRequired(ConnectionProperty<V, T> property)
+    {
+        return property.getRequiredValue(properties);
+    }
+
+    private <V, T> Optional<T> resolveOptional(ConnectionProperty<V, T> property)
+    {
+        return property.getValue(properties);
+    }
+
+    private <V, T> T resolveWithDefault(ConnectionProperty<V, T> property, T defaultValue)
+    {
+        return property.getValueOrDefault(properties, defaultValue);
+    }
+
+    public static Properties urlProperties(String url, Properties info)
```

Comment:
What does it do exactly? doc?

Reply:
Moved to TrinoDriver (it's used there only)